Add delta option to the scp command, using rsync

afbjorklund opened the pull request:
When doing repeated syncs, or copying large directories, it is more
efficient to use "rsync" than to use "scp". Both use "ssh" to do it.

afbjorklund commented:
Apparently rsync is not in the Travis default tool belt. \<sigh\>

nathanleclaire commented:
Ah cool, have long thought this would be a nice feature to have.

nathanleclaire commented:
I like it -- how about `--delta` or `--diff` for the flag name instead? Any objections, or other ideas? (`--update` is just so general, and `--rsync` short form of `-r` would already be taken.)

afbjorklund commented:
@nathanleclaire : OK, I changed the option to `--delta` and removed the `-u` flag...

It wasn't all that useful anyway, since rsync will still use the "quick check" on the files.
I avoided the name --diff, since it reminds me of diff(1). And --rsync was more a detail.

afbjorklund commented:
Aargh.

nathanleclaire commented:
Looks epic! Awesome, I've wanted this for quite a while

nathanleclaire commented:
Could you also turn in a PR to https://github.com/docker/docker.github.io updating the `scp` command documentation in `vnext-machine` branch?

LGTM

afbjorklund commented:
will do!

@nathanleclaire : as always with rsync, it's a bit peculiar when it comes to directories... (`-u`)
The "safe" option is to use trailing slash `'/'` on both source and destination to avoid surprises

i.e. /tmp/foo -> /tmp/foo, and you might end up with /tmp/foo/foo or something.

nathanleclaire commented:
> @nathanleclaire : as always with rsync, it's a bit peculiar when it comes to directories... (-r)
The "safe" option is to use trailing slash '/' on both source and destination to avoid surprises

Interesting, perhaps something to note in the documentation!

afbjorklund commented:
Yeah, that is what I thought. It should come as no surprise to rsync users.
But maybe it can trip someone up when switching from scp or something.
